src/metal.cs

```diff
@@ -2892,8 +3173,158 @@ interface MTLTileRenderPipelineDescriptor : NSCopying {
 		[Export ("tileBuffers")]
 		MTLPipelineBufferDescriptorArray TileBuffers { get; }
 
+		[NoMac, iOS (12,0), TV (12,0)]
+		[Export ("maxTotalThreadsPerThreadgroup")]
+		nuint MaxTotalThreadsPerThreadgroup { get; set; }
+
 		[Export ("reset")]
 		void Reset ();
 	}
+
+	interface IMTLEvent {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[Protocol]
+	interface MTLEvent {
+		[Abstract]
+		[NullAllowed, Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+	}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	[DesignatedDefaultCtor]
+	interface MTLSharedEventListener {
+		[Export ("initWithDispatchQueue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (DispatchQueue dispatchQueue);
+
+		[Export ("dispatchQueue")]
+		DispatchQueue DispatchQueue { get; }
+	}
+
+	delegate void MTLSharedEventNotificationBlock (IMTLSharedEvent arg0, ulong arg1);
```

Comment:
sharpie lost the argument names, but they should be in the headers

Reply:
`typedef void (^MTLSharedEventNotificationBlock)(id <MTLSharedEvent>, uint64_t value);` I guess `event` and `value`.